Kernel/Devices/USB/USBPipe.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Devices/USB/PacketTypes.h>
+#include <Kernel/Devices/USB/UHCIController.h>
+#include <Kernel/Devices/USB/USBPipe.h>
+#include <Kernel/Devices/USB/USBTransfer.h>
+
+namespace Kernel::USB {
+
+OwnPtr<USBPipe> USBPipe::create_pipe(Type type, Direction direction, u8 endpoint_address, u16 max_packet_size, i8 device_address, u8 poll_interval)
+{
+    return make<USBPipe>(type, direction, endpoint_address, max_packet_size, device_address, poll_interval);
```

Comment:
Same OOM suggestion (`make<T>` assumes success), but again I'm not positive this can be called passed boot.

Reply:
```suggestion
    return make<USBPipe>(type, direction, endpoint_address, max_packet_size, device_address, poll_interval);
```

```suggestion
    auto pipe = adopt_own_if_nonnull(new USBPipe(type, direction, endpoint_address, max_packet_size, device_address, poll_interval));
    if (!pipe)
        return { } ;

    return pipe;
```

Perhaps this?